File: php/source.php
<?php
// Include the Postgres connection details
require 'connection.php';

$sql = "CREATE SOURCE counter FROM LOAD GENERATOR COUNTER";

$statement = $connection->prepare($sql);
$statement->execute();

$sources = "SHOW SOURCES";
$statement = $connection->query($sources);
$result = $statement->fetchAll(PDO::FETCH_ASSOC);
var_dump($result);

File: rust/src/source.rs
use postgres::Error;

use crate::connection::create_client;

/// Creates a load generator source
pub(crate) fn create_source() -> Result<u64, Error> {
    let mut client = create_client().expect("Error creating client.");

    client.execute(
        "
        CREATE SOURCE IF NOT EXISTS counter
        FROM LOAD GENERATOR COUNTER
        (TICK INTERVAL '500ms')
        WITH (SIZE = '3xsmall');
    ",
        &[],
    )
}
